myokit/formats/sbml/_importer.py

```diff
@@ -96,53 +102,76 @@ def model(self, path):
         refs = {}
 
         # Handle notes, if given
-        x = dom_child(xmodel, 'notes')
+        x = xmodel.find(self.ns['sbml'] + 'notes')
         if x:
             log.log('Converting <model> notes to ascii')
-            model.meta['desc'] = html2ascii(x.toxml(), width=75)
+            model.meta['desc'] = html2ascii(ET.tostring(x, encoding='unicode'),
+                                            width=75
+                                            )
             # width = 79 - 4 for tab!
 
         # Warn about missing functionality
-        x = dom_child(xmodel, 'listOfCompartments')
+        x = xmodel.find(self.ns['sbml'] + 'listOfCompartments')
         if x:   # pragma: no cover
             log.warn('Compartments are not supported.')
-        x = dom_child(xmodel, 'listOfSpecies')
+        x = xmodel.find(self.ns['sbml'] + 'listOfSpecies')
         if x:   # pragma: no cover
             log.warn('Species are not supported.')
-        x = dom_child(xmodel, 'listOfConstraints')
+        x = xmodel.find(self.ns['sbml'] + 'listOfConstraints')
         if x:   # pragma: no cover
             log.warn('Constraints are not supported.')
-        x = dom_child(xmodel, 'listOfReactions')
+        x = xmodel.find(self.ns['sbml'] + 'listOfReactions')
         if x:   # pragma: no cover
             log.warn('Reactions are not supported.')
-        x = dom_child(xmodel, 'listOfEvents')
+        x = xmodel.find(self.ns['sbml'] + 'listOfEvents')
         if x:   # pragma: no cover
             log.warn('Events are not supported.')
 
         # Ignore custom functions
-        x = dom_child(xmodel, 'listOfFunctionDefinitions')
+        x = xmodel.find(self.ns['sbml'] + 'listOfFunctionDefinitions')
         if x:   # pragma: no cover
             log.warn('Custom math functions are not (yet) implemented.')
 
         # Parse custom units
-        x = dom_child(xmodel, 'listOfUnitDefinitions')
+        x = xmodel.find(self.ns['sbml'] + 'listOfUnitDefinitions')
         if x:
-            self._parse_units(model, comp, x)
+            self._parse_units(model, comp, x, self.ns['sbml'])
+
+        # add time as independent variable (not explicit in SBML format)
+        if bind_time:
+            # add and bind time variable to component
+            comp.add_variable('time')
+            time = comp.get('time')
+            time.set_binding('time')
+            # set unit and value
+            if 'time' in self.units.keys():
```

Comment:
A slightly more efficient way to do this is:
```
try:
    unit = self.units['time']
except KeyError:
    unit = myokit.units.s
    log.warn(...)
```
This version only has a single dict lookup (and some python fans prefer it for more philosophical reasons too)

Reply:
Is also more elegant. Done.